pkg/provider/kubernetes/ingress/kubernetes.go

```diff
@@ -187,12 +187,11 @@ func (p *Provider) loadConfigurationFromIngresses(ctx context.Context, client Cl
 	serverVersion, err := client.GetServerVersion()
 	if err != nil {
 		log.FromContext(ctx).Errorf("Failed to get server version: %v", err)
-		return conf
 	}
 
 	var ingressClasses []*networkingv1beta1.IngressClass
 
-	if supportsIngressClass(serverVersion) {
+	if serverVersion != nil && supportsIngressClass(serverVersion) {
```

Comment:
I think you can move this into the `supportsIngressClass()` method

Reply:
Good call.